### DISCUSSION TITLE
Thanks for such a great tool!

### DISCUSSION THREAD

**Discussion by nakdeyes** [cajhin/capsicain#79] [General]
Just wanted to reach out and say thanks so much for making this tool and then sharing it with the world. 

I had been hobbling by with Windows Power Toys keyboard remapping for my CTRL <-> CAPS swap. It worked in like .. ~80% of places so I just kinda lived with the other 20% of games and RDP software where it didn't work ( and then would cause issues! So was constantly enabling / disabling ). 

I also tried the Windows Keyboard Layout Creator, but it seemed like they just couldn't remap CAPS / CTRL? Lame! 

Then I found your tool <3. It was easy to setup and seems to be low level enough that all my stuff works with it so far! 

Just wanted to say THANK YOU and keep up the good work! 

Trying to think if I could contribute.. might be fun to make a config file UI so you could remap in a UI instead of a config file? Might be cool. Cheers!

**Comment by cajhin** [comment]
Thanks for the flowers :)

I believe a GUI with at least the basic config options would be very helpful for beginners; it's all a bit cryptic, I must admit (the ini style turned out to be optimized for a power user with a lot of configuration rules).

Downsides:
- it's effort and I don't need it myself
- GUI needs to be maintained when the ini spec changes or new features are added (but that didn't happen in a long time, it's rather stable now)
- I would never code GUI stuff in C, C# would be much nicer. (I actually toyed with a complete capsicain rewrite in C# as a Windows Forms app, but lost interest). JavaScript in the browser could be nice, although I don't like the language.
- I'm not overly proud of the oldschool .ini style (but I don't want to deal with a huge yaml config file either)

Mostly, the lack of a GUI is because I never came up with a convincing, easy to understand GUI design.

If you'd like to tackle that, you're welcome. IMO it doesn't need to be feature complete, but what is there needs to be flawless (capsicain is complicated enough without bugs on top).
